docs/animation/armature.md

```diff
@@ -0,0 +1,183 @@
+Armature Files
+==============
+
+Armature Files define joints and clips for animating a model.
+
+File Structure
+----------------
+
+An example armature file, taken from the forge debug mod (requires #3875)
+```json
+{
+  "joints": {
+    "stick": {"2": [1.0]},
+    "cube": {"3": [1.0]}
+  },
+  "clips": {
+    "default": {
+      "loop": true,
+      "joint_clips": {
+        "stick": [
+          {
+            "variable": "offset_x",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 0.6875, 0]
+          }
+        ],
+        "cube": [
+          {
+            "variable": "offset_x",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 0.6875, 0]
+          },
+          {
+            "variable": "axis_z",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 1 ]
+          },
+          {
+            "variable": "origin_x",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 0.15625 ]
+          },
+          {
+            "variable": "origin_y",
+            "type": "uniform",
+            "interpolation": "nearest",
+            "samples": [ 0.40625 ]
+          },
+          {
+            "variable": "angle",
+            "type": "uniform",
+            "interpolation": "linear",
+            "samples": [0, 120, 240, 0, 120, 240]
+          }
+        ]
+      },
+      "events": {}
+    }
+  }
+}
+
+```
+
+The file is organized in two sections, joints and clips.
+
+Joints
+--------
+Each joint defines a connection to the model file. A joint can animate any number of elements in the model file, but they are all animated with the same input value and transforms
+
+The format is like this:
+```javascript
+ {
+    "joints": {
+        <joint>, ...
+    }
+}
+    
+---
+
+<joint> ::= {
+    <string>: {  // joint name
+        <joint_definition>, ...
+    }
+}
+
+<joint_definition> ::= {
+    <string>: [ <float> ] // index_model, joint_weight
+}
+
+```
+
+- `joint_name` is the name of the joint
+- `index_model` is a 0-indexed number denoting a model element this joint controls. Must be a string (see example)
+- `joint_weight` is a weight (0-1) of how much this element will animate. The transformation matrix caused by a clip animating this element will be multiplied by this number.
```

Comment:
according to fry a while ago the weights are unused, is that still the case?

also by "transformation matrix will be multiplied by this number" what does that mean, the transform is lerped, or the whole matrix is just scalarly multiplied, or what?

Reply:
i remember seeing it being used it ModelBlockAnimation, lines 545-570.